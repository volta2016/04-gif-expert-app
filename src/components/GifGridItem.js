import React from 'react'
import PropTypes from 'prop-types'


const GifGridItem = ({ title, url }) => {

    // console.log({id, title, url});

    return (
        
        <div className="card animate__animated animate__fadeIn">
          <img src={url} alt={title} />
          <p>{title}</p>
        </div>
    
        
    )
}



GifGridItem.proTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired

}

export default GifGridItem;
