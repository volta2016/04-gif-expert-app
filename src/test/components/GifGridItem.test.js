
import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import GifGridItem  from '../../components/GifGridItem';


describe('Pruebas en <GifGridItem />', () => {

  const title = 'Un TÃ­tulo';
  const url = 'https://localhost/algo.jpg';
  const wrapper  = shallow(<GifGridItem title={title} url={url}/>)
  
  test('debe de mostrar el componente correctamente', () => { 
    expect(wrapper).toMatchSnapshot(); 
  })

  test('debe de tener un parrafo con el title', () => {
    
    const paragraph = wrapper.find('p');
    expect(paragraph.text().trim()).toBe(title)

  })

  test('debe de tener la imagen a la url de los props', () => {

    const img = wrapper.find('img');
    // console.log(img.prop('src'));
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  })
  
  
  
});

