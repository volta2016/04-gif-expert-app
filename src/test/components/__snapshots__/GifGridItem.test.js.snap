// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un Título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un Título
  </p>
</div>
`;
